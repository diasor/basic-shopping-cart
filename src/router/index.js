import Vue from "vue"
import VueRouter from "vue-router"
import Home from "@views/Home.vue"

Vue.use(VueRouter)

const routes = [
  {
	path: "/error",
	name: 'Error',
	component: () => import(/* webpackChunkName: "about" */ "../views/Error.vue")
  },
  {
    path: "/",
    name: "Home",
    component: Home
  },
  {
	path: "*",
	redirect: '/error'
  }
]

const router = new VueRouter({
	mode: "history",
	routes
})

export default router
