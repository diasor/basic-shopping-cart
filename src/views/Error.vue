<template>
	<v-card
		class="mx-auto mt-10"
		max-width="80%"
		justify="center"
		data-testid="error-card"
	>

	<v-card-title>
		Ooops! An error has occured.
	</v-card-title>

	<v-card-subtitle>
		Please try again.
	</v-card-subtitle>
	<v-card-actions>
		<v-btn
			data-testid="back-home-button"
			color="#8d010d lighten-2"
			text
			@click="backHome()"
		>
			Try Again
			<v-icon class="ml-4">fas fa-circle-notch fa-spin</v-icon>
		</v-btn>
    </v-card-actions>

  </v-card>
</template>
<script>
export default {
	name: "Error",

	methods: {
		backHome () {
			this.$router.push({ path: '/' })
		}
	}
}
</script>
