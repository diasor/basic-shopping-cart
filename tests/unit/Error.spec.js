import { mount, createLocalVue } from "@vue/test-utils"
import Error from "@views/Error.vue"
import Vuetify from "vuetify"
import router from "@router"

let localVue
let wrapper
let vuetify

afterEach(() => {
	wrapper.destroy()
})

describe("testing Error Template", () => {
	beforeEach(() => createInstance())
	it("should render the page and match the snapshot", () => {
		expect(wrapper.exists()).toBe(true)
		expect(wrapper).toMatchSnapshot()
	})

	it("should render a v-card with the proper test name", () => 
		expect(wrapper.find(".v-card[data-testid=\"error-card\"]").exists()).toBe(true))

	it("should render a card title with the proper content", () => {
		const title = wrapper.find(".v-card__title")
		expect(title.exists()).toBe(true)
		expect(title.text()).toBe("Ooops! An error has occured.")
	})

	it("should render a card sub title with the proper content", () => {
		const title = wrapper.find(".v-card__subtitle")
		expect(title.exists()).toBe(true)
		expect(title.text()).toBe("Please try again.")
	})

	it("should render a Button component with the correct name and props", () => {
		const button = wrapper.find(".v-btn[data-testid=\"back-home-button\"]")
		expect(button.exists()).toBe(true)
		expect(button.text()).toBe("Try Again")
		expect(button.props().color).toBe("#8d010d lighten-2")
	})
})

describe("testing Error Methods", () => {
	beforeEach(() => createInstance())

	describe("testing backHome method", () => {
		it("should emit onSubmit", async() => {
			wrapper.vm.backHome()
			await wrapper.vm.$nextTick()
			expect(wrapper.vm.$route.name).toEqual("Home")
		})
	})
})

/**
 * [createInstance is a function that defines all the boiler plate code to create the Error component ]
 */
async function createInstance() {
	localVue = createLocalVue()
	vuetify = new Vuetify()

	// mount the wrapper
	wrapper = mount(Error, {
		localVue,
		vuetify,
		router,
	})
	await wrapper.vm.$nextTick()
}