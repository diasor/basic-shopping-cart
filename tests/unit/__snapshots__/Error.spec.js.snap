// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Error Template should render the page and match the snapshot 1`] = `
<div class="mx-auto mt-10 v-card v-sheet theme--light" style="max-width: 80%;" justify="center" data-testid="error-card">
  <div class="v-card__title">
    Ooops! An error has occured.
  </div>
  <div class="v-card__subtitle">
    Please try again.
  </div>
  <div class="v-card__actions"><button type="button" class="v-btn v-btn--text theme--light v-size--default" style="caret-color: #8d010d lighten-2;" data-testid="back-home-button"><span class="v-btn__content">
			Try Again
			<i aria-hidden="true" class="v-icon notranslate ml-4 fas fa-circle-notch fa-spin theme--light"></i></span></button></div>
</div>
`;
